modules/llrt_os/src/lib.rs

```diff
@@ -1,47 +1,291 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-use std::env;
+use std::{
+    collections::HashMap,
+    env,
+    net::{Ipv4Addr, Ipv6Addr},
+    sync::{Arc, Mutex},
+    thread,
+};
 
 use llrt_utils::{
     module::{export_default, ModuleInfo},
+    result::ResultExt,
     sysinfo::{get_arch, get_platform},
 };
+use once_cell::sync::Lazy;
 use rquickjs::{
     module::{Declarations, Exports, ModuleDef},
     prelude::Func,
-    Ctx, Result,
+    Ctx, Exception, Object, Result,
 };
+use sysinfo::{CpuRefreshKind, MemoryRefreshKind, Networks, RefreshKind, System};
 
 #[cfg(unix)]
-use self::unix::{get_release, get_type, get_version, EOL};
+use self::unix::{get_priority, get_type, get_user_info, get_version, set_priority, DEV_NULL, EOL};
 #[cfg(windows)]
-use self::windows::{get_release, get_type, get_version, EOL};
+use self::windows::{
+    get_priority, get_type, get_user_info, get_version, set_priority, DEV_NULL, EOL,
+};
 
 #[cfg(unix)]
 mod unix;
 #[cfg(windows)]
 mod windows;
 
+static SYSTEM: Lazy<Arc<Mutex<System>>> = Lazy::new(|| {
+    Arc::new(Mutex::new(System::new_with_specifics(
+        RefreshKind::new()
+            .with_cpu(CpuRefreshKind::new().with_cpu_usage().with_frequency())
+            .with_memory(MemoryRefreshKind::new().with_ram()),
+    )))
+});
+
+static NETWORKS: Lazy<Arc<Mutex<Networks>>> =
+    Lazy::new(|| Arc::new(Mutex::new(Networks::new_with_refreshed_list())));
```

Comment:
There is medium to high a chance that at some point the code that update those mutexes will panic because we call a lot of C code in it. If that happens those mutexes will be poisoned for all future executions even if that particular quickjs is killed. That is why I usually don't use the std mutexes and I prefer the parking_lot mutexes.
@richarddavison What are your thoughts on that

Reply:
Indeed, parking_lot::Mutex looks good, so I installed it.  Please check it out. :)

Reply:
@Sytten I don't see a lot of c code in the sysinfo create. It basically reads unix files that contains system info mostly in pure rust. Nevertheless I see some nice benefits of using parking lot mutex instead of std one:

> - No poisoning, the lock is released normally on panic.
> - Only requires 1 byte of space, whereas the standard library boxes the Mutex due to platform limitations.
> - Can be statically constructed.
> - Does not require any drop glue when dropped.
> - Inline fast path for the uncontended case.
> - Efficient handling of micro-contention using adaptive spinning.
> - Allows raw locking & unlocking without a guard.
> - Supports eventual fairness so that the mutex is fair on average.
> - Optionally allows making the mutex fair by calling MutexGuard::unlock_fair.

Similarly we have this for RwLock from parking_lot. If we add this we should use this crate exclusively for all RwLocks and Mutexes. But this is out of scope for this PR. So @nabetti1720 please use std mutex and (if you want) create a new PR replacing all Mutexes and RwLocks :)

Reply:
Sorry, I was a bit premature. I have cut back to `std::sync::Mutex` and would appreciate confirmation. :)